File: backend/demo/src/main/java/com/example/demo/service/QuestionService.java
package com.example.demo.service;

import java.util.List;

import org.aspectj.weaver.patterns.TypePatternQuestions.Question;

import com.example.demo.model.Questions;

public interface QuestionService {

	void addQuestion(Questions question, int tid);

	List<Questions> getAllQuestions();

	void updateQuestion(Questions question);

	void deleteQuestion(int qid, int tid);

	

}

File: frontend/examFront/src/app/question.service.ts
import { Injectable } from '@angular/core';
import{HttpClient,HttpClientModule,HttpHeaders } from '@angular/common/http';
import{Observable} from 'rxjs';
import { Question } from "src/app/Model/Question.model";


@Injectable({
  providedIn: 'root'
})
export class QuestionService {

constructor(private http: HttpClient) { }

addQuestion(question:Question,tid:number):Observable<any> {

  return this.http.post(`http://localhost:8080/addQuestion/${tid}`, question);
}

deleteQuestion(qid:number,tid:number):Observable<any> {
  return this.http.delete(`http://localhost:8080/deleteQuestion/${qid}/${tid}`);
}

updateQuestion(question:Question):Observable<any>{
  return this.http.put(`http://localhost:8080/updateQuestion`, question);
}
}
